### PR TITLE
MB-15671 Replace MakeMove helpers

### PR DESCRIPTION
**Pull request by mlm55**
## Summary

This ticket replaces some of the `Make*MoveWithShipment` helper functions. Since we no longer use selectedMoveType, most of these can be consolidated. The old `MakeHHGMoveWithShipment` allowed move assertions to be passed in, and the new `BuildMoveWithShipment` function calls `BuildMTOShipmentWithMove` (which doesn't allow move customizations). So we added a new function to remove the customizations of a specific type from the customs slice. 

## Setup to Run Your Code

1. Run `make server_test` and make sure tests pass
2. Run `make db_dev_e2e_populate` and make sure the db populates correctly

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/services/move_task_order/move_task_order_updater_test.go [hunk 1]
```diff
@@ -29,11 +29,18 @@ func (suite *MoveTaskOrderServiceSuite) TestMoveTaskOrderUpdater_UpdateStatusSer
...
 
 	suite.Run("Move status is updated successfully (with HHG shipment)", func() {
-		move := testdatagen.MakeHHGMoveWithShipment(suite.DB(), testdatagen.Assertions{
-			Move: models.Move{
-				Status: models.MoveStatusNeedsServiceCounseling,
+		//move := testdatagen.MakeHHGMoveWithShipment(suite.DB(), testdatagen.Assertions{
+		//	Move: models.Move{
+		//		Status: models.MoveStatusNeedsServiceCounseling,
+		//	},
+		//})
```
Comment (ahobson) [inline pkg/services/move_task_order/move_task_order_updater_test.go:36]: Maybe we don't need to keep the commented out bits?

File: pkg/factory/move_factory_test.go [hunk 2]
```diff
@@ -252,5 +252,45 @@ func (suite *FactorySuite) TestBuildMove() {
 		suite.Equal(availableToPrimeAt, *move.AvailableToPrimeAt)
 		suite.NotNil(move.AvailableToPrimeAt)
 	})
+	suite.Run("Successful creation of move with shipment", func() {
+		// Under test:      BuildMoveWithShipment
+		// Set up:          Create a move using BuildMoveWithShipment
+		// Expected outcome:Move with shipment
+
+		move := BuildMoveWithShipment(suite.DB(), nil, nil)
+		suite.NotNil(move.MTOShipments)
```
Comment (ahobson) [inline pkg/factory/move_factory_test.go:261]: Huh, I was thinking this should be `suite.NotEmpty` ?

File: pkg/services/payment_request/payment_request_list_fetcher_test.go [hunk 3]
```diff
@@ -464,39 +467,47 @@ func (suite *PaymentRequestServiceSuite) TestListPaymentRequestWithSortOrder() {
...
-		expectedMove1 := testdatagen.MakeHHGMoveWithShipment(suite.DB(), testdatagen.Assertions{
-			ServiceMember: models.ServiceMember{
-				Edipi:       models.StringPointer("EZFG"),
-				LastName:    models.StringPointer("Spacemen"),
-				FirstName:   models.StringPointer("Lena"),
-				Affiliation: &branchNavy,
-			},
-			Move: models.Move{
-				Locator: "AAAA",
+		// ca
```
Comment (ahobson) [inline pkg/services/payment_request/payment_request_list_fetcher_test.go:470]: copy and paste error?

File: pkg/services/payment_request/payment_request_list_fetcher_test.go [hunk 4]
```diff
@@ -464,39 +467,46 @@ func (suite *PaymentRequestServiceSuite) TestListPaymentRequestWithSortOrder() {
...
+		expectedMove1 := factory.BuildMoveWithShipment(suite.DB(), []factory.Customization{
+			{
+				Model: models.ServiceMember{
+					Edipi:       models.StringPointer("EZFG"),
+					LastName:    models.StringPointer("Spacemen"),
+					FirstName:   models.StringPointer("Lena"),
+					Affiliation: &branchNavy,
+				},
 			},
-			PaymentRequest: models.PaymentRequest{
```
Comment (mlm55) [inline pkg/services/payment_request/payment_request_list_fetcher_test.go:477]: i don't think this MakeHHGMoveWithShipment did anything with this assertion, so i removed it

Comment (ahobson) [reply pkg/services/payment_request/payment_request_list_fetcher_test.go:477]: I agree. I noticed something similar elsewhere

**Review by ahobson — APPROVED** [review]
🥇 Great stuff!

**Pull request merged**